syntax = "proto3";

service MovingAverageService {
  rpc CalculateMovingAverage(MovingAverageRequest) returns (MovingAverageResponse) {}
}

message MovingAverageRequest {
  repeated float stock_data = 1;
  int32 window_size = 2;
}

message MovingAverageResponse {
  repeated double values = 1;
}

// python -m grpc_tools.protoc -I src/calculate_indicator/moving_average --python_out src/calculate_indicator/moving_average --grpc_python_out src/calculate_indicator/moving_average src/calculate_indicator/moving_average/calculate_indicator.proto